src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/DeterministicMethodAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DeterministicMethodAnalyzer : DiagnosticAnalyzer
+    {
+        private OrchestratorMethodCollector orchestratorMethodCollector;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return ImmutableArray.Create(
+                    DateTimeAnalyzer.Rule,
+                    EnvironmentVariableAnalyzer.Rule,
+                    GuidAnalyzer.Rule,
+                    IOTypesAnalyzer.Rule,
+                    ThreadTaskAnalyzer.Rule,
+                    TimerAnalyzer.V1Rule,
+                    TimerAnalyzer.V2Rule,
+                    CancellationTokenAnalyzer.Rule,
+                    MethodInvocationAnalyzer.Rule);
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            orchestratorMethodCollector = new OrchestratorMethodCollector();
+
+            context.RegisterCompilationStartAction(compilation =>
+            {
+                compilation.RegisterSyntaxNodeAction(orchestratorMethodCollector.FindOrchestratorMethods, SyntaxKind.MethodDeclaration);
+
+                compilation.RegisterCompilationEndAction(RegisterAnalyzers);
+            });
+        }
+
+        private void RegisterAnalyzers(CompilationAnalysisContext context)
+        {
+            var methodInvocationAnalyzer = new MethodInvocationAnalyzer();
+            var orchestratorMethods = orchestratorMethodCollector.GetOrchestratorMethods();
+            Parallel.ForEach(orchestratorMethods, methodInformation =>
```

Comment:
You don't necessarily need to change this, but I think it's interesting that we're running all these analyzers in parallel. I'm guessing it makes the analysis run faster but I wonder what the CPU impact would be to allocate and run a new thread for each orchestrator method.

Reply:
I've reverted them back to for each loops. I suppose if someone is actively typing I don't want them to have a poor experience with Durable, just in case it would affect their experience. So it's worth it to be safe here, rather than wait for someone to complain.